boefjes/boefjes/api.py

```diff
@@ -0,0 +1,182 @@
+import base64
+import logging
+import multiprocessing
+from datetime import datetime, timezone
+from enum import Enum
+from typing import List, Optional
+
+from fastapi import Depends, FastAPI, HTTPException, Response
+from pydantic import BaseModel, Extra, Field
+from requests import HTTPError
+from uvicorn import Config, Server
+
+from boefjes.clients.bytes_client import BytesAPIClient
+from boefjes.clients.scheduler_client import SchedulerAPIClient, TaskStatus
+from boefjes.config import settings
+from boefjes.job_handler import (
+    _collect_default_mime_types,
+    _find_ooi_in_past,
+    get_environment_settings,
+    get_octopoes_api_connector,
+    serialize_ooi,
+)
+from boefjes.job_models import BoefjeMeta
+from boefjes.katalogus.local_repository import LocalPluginRepository, get_local_repository
+from octopoes.models import Reference
+
+app = FastAPI(title="Boefje API")
+logger = logging.getLogger(__name__)
+
+
+class UvicornServer(multiprocessing.Process):
+    def __init__(self, config: Config):
+        super().__init__()
+        self.server = Server(config=config)
+        self.config = config
+
+    def stop(self):
+        self.terminate()
+
+    def run(self, *args, **kwargs):
+        self.server.run()
+
+
+def run():
+    config = Config(app, host=settings.boefje_api_host, port=settings.boefje_api_port)
+    instance = UvicornServer(config=config)
+    instance.start()
+    return instance
+
+
+class BoefjeInput(BaseModel):
+    task_id: str
+    output_url: str
+    boefje_meta: BoefjeMeta
+
+    class Config:
+        extra = Extra.forbid
+
+
+class StatusEnum(str, Enum):
+    COMPLETED = "COMPLETED"
+    FAILED = "FAILED"
+
+
+class File(BaseModel):
+    name: Optional[str]
+    content: str = Field(..., contentEncoding="base64")
+    tags: Optional[List[str]]
+
+
+class BoefjeOutput(BaseModel):
+    status: StatusEnum
+    files: Optional[List[File]]
+
+
+def get_scheduler_client():
+    return SchedulerAPIClient(settings.scheduler_api)
+
+
+def get_bytes_client():
+    return BytesAPIClient(
+        settings.bytes_api,
+        username=settings.bytes_username,
+        password=settings.bytes_password,
+    )
+
+
+@app.get("/healthz")
+async def root():
+    return "OK"
+
+
+@app.get("/api/v0/tasks/{task_id}", response_model=BoefjeInput)
+async def boefje_input(
+    task_id: str,
+    scheduler_client: SchedulerAPIClient = Depends(get_scheduler_client),
+    local_repository: LocalPluginRepository = Depends(get_local_repository),
+):
+    task = get_task(task_id, scheduler_client)
```

Comment:
I didn't realize this when doing the first review, but doing sync requests like this will block the asyncio loop as far as I understand it. I don't think this is a blocker, but we should probably make the scheduler client also async in the future.

Reply:
This is indeed not ideal, but we'd have to create new versions of the scheduler, bytes, octopoes and katalogus API clients to make them async. If we refactor the current API clients, we'd also have to change all existing usages to await the async calls.